Tests/Unit/Office365DSC/Office365DSC.SPOHomeSite.Tests.ps1

```diff
@@ -0,0 +1,161 @@
+
+            Mock -CommandName Remove-PnPHomeSite -MockWith {
+
+            }
+
+            It "Should return present from the Get method" {
+                (Get-TargetResource @testParams).Ensure | Should Be "Present"
+            }
+
+            It "Should return false from the Test method" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "Should call Remove-PnPHomeSite" {
+                Set-TargetResource @testParams
+                Assert-MockCalled  Remove-PnPHomeSite
+            }
+        }
+
+        Context -Name "When there should be a home site set and there is not or it's the wrong one" -Fixture {
+            $testParams = @{
+                IsSingleInstance   = "Yes"
+                Url                = "https://contoso.sharepoint.com/sites/homesite"
+                GlobalAdminAccount = $GlobalAdminAccount
+                Ensure             = "Present"
+            }
+
+            Mock -CommandName Get-SPOHomeSite -MockWith {
+                return "https://contoso.sharepoint.com/sites/homesite"
+            }
+
+            Mock -CommandName Get-SPOSite -MockWith {
+                throw
+            }
+
+            Mock -CommandName Set-PnPHomeSite -MockWith {
+            }
+
+
+            Mock -CommandName New-Office365DSCLogEntry -MockWith {
+            }
+
+
+            It "Should return Present from the Get method" {
+                (Get-TargetResource @testParams).Ensure | Should Be "Present"
+            }
+
+            It "Should return false from the Test method" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "Should throw an error" {
+                { Set-TargetResource @testParams } | Should Throw "The specified Site Collection $($testParams.Url) for SPOHomeSite doesn't exist."
+                Assert-MockCalled Get-SPOSite
+                Assert-MockCalled New-Office365DSCLogEntry
+            }
+        }
+
+        Context -Name "It should set the home site" -Fixture {
+            $testParams = @{
+                IsSingleInstance   = "Yes"
+                Url                = "https://contoso.sharepoint.com/sites/homesite"
+                GlobalAdminAccount = $GlobalAdminAccount
+                Ensure             = "Present"
+            }
+
+            Mock -CommandName Get-SPOHomeSite -MockWith {
+                return "https://contoso.sharepoint.com/sites/homesite1"
+            }
+
+            Mock -CommandName Set-PnPHomeSite -MockWith {
+            }
+
+            Mock -CommandName Get-SPOSite -MockWith {
+
+            }
+
+            It "Should set the correct site" {
+                Set-TargetResource @testParams
+                Assert-MockCalled Get-SPOSite
+                Assert-MockCalled Set-PnPHomeSite
+            }
+        }
+
+        Context -Name "ReverseDSC Tests" -Fixture {
+            $testParams = @{
+                IsSingleInstance   = "Yes"
```

Comment:
Again, Export-TargetResource should only ever define GlobalAdminAccount as parameters.